pkg/api/meta/meta.go

```diff
@@ -359,6 +359,9 @@ func extractFromOwnerReference(v reflect.Value, o *metatypes.OwnerReference) err
 	if err := runtime.Field(v, "UID", &o.UID); err != nil {
 		return err
 	}
+	if err := runtime.Field(v, "Controller", &o.Controller); err != nil {
```

Comment:
This will be a shallow copy of the pointer. Could you add a line before `return nil` to make it a deep copy?